### PR TITLE
git-sim: init at 0.3.5

### PR DESCRIPTION
**Pull request by MathiasSven**
Closes: #212680 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  installShellFiles,
+  gitpython,
+  typer,
+  pydantic-settings,
+}:
+
+buildPythonPackage rec {
+  pname = "git-dummy";
+  version = "0.1.2";
+  format = "setuptools";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-dummy/default.nix:14]:
```suggestion
  pyproject = true;
```
please use the new style described in language-frameworks/python.

You probably need to add setuptools to build-system

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  src = fetchFromGitHub {
+    owner = "initialcommit-com";
+    repo = "git-dummy";
+    rev = "v${version}";
+    hash = "sha256-viybxn2J7SO7NgSvjwlP+tgtm+H6QrACafIy82d9XEk=";
+  };
+
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-dummy/default.nix:23]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    pydantic-settings
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    for shell in bash fish zsh; do
+      $out/bin/git-dummy --show-completion $shell > git-dummy.$shell
+      installShellCompletion git-dummy.$shell
+    done
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-dummy/default.nix:35]:
```suggestion
    installShellCompletion --cmd git-dummy \
      --bash <($out/bin/git-dummy --show-completion bash)
      --fish <($out/bin/git-dummy --show-completion fish)
      --zsh <($out/bin/git-dummy --show-completion zsh)
```

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/initialcommit-com/git-dummy";
+    description = "Generate dummy Git repositories populated with the desired number of commits, branches, and structure.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-dummy/default.nix:40]:
```suggestion
    description = "Generate dummy Git repositories populated with the desired number of commits, branches, and structure";
```

File: pkgs/development/python-modules/git-sim/default.nix [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  typer,
+  pydantic,
+  fonttools,
+  git-dummy,
+}:
+
+buildPythonPackage rec {
+  pname = "git-sim";
+  version = "0.3.5";
+  format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-sim/default.nix:18]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/git-sim/default.nix [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  # https://github.com/NixOS/nixpkgs/commit/8033561015355dd3c3cf419d81ead31e534d2138
+  makeWrapperArgs = [ "--prefix PYTHONWARNINGS , ignore:::pydub.utils:" ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    for shell in bash fish zsh; do
+      $out/bin/git-sim --show-completion $shell > git-sim.$shell
+      installShellCompletion git-sim.$shell
+    done
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-sim/default.nix:50]: Can you do the same as above?

File: pkgs/development/python-modules/git-sim/default.nix [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    ln -s ${git-dummy}/bin/git-dummy $out/bin/
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-sim/default.nix:52]: Why not put that into the wrapper under the PATH env?

File: pkgs/development/python-modules/git-sim/default.nix [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  src = fetchFromGitHub {
+    owner = "initialcommit-com";
+    repo = "git-sim";
+    rev = "v${version}";
+    hash = "sha256-4jHkAlF2SAzHjBi8pmAJ0TKkcLxw+6EdGsXnHZUMILw=";
+  };
+
+  postPatch = ''
+    sed -i '/opencv-python-headless/d' pyproject.toml
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/git-sim/default.nix:29]: Please use the relaxDepsHook instead

File: pkgs/development/python-modules/git-sim/default.nix (hunk 7 above)
Comment (MathiasSven) [reply pkgs/development/python-modules/git-sim/default.nix:50]: I am not sure what you mean in this case, would `git-dummy` still be available in the user's `PATH`? `git-sim` itself doesn't need `git-dummy` in its `PATH`, it only uses it for tests, and even then it uses it as a module which is covered by `dependencies`.

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      --bash <($out/bin/git-dummy --show-completion bash) \
+      --fish <($out/bin/git-dummy --show-completion fish) \
+      --zsh <($out/bin/git-dummy --show-completion zsh)
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/initialcommit-com/git-dummy";
+    description = "Generate dummy Git repositories populated with the desired number of commits, branches, and structure";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ mathiassven ];
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/git-dummy/default.nix:45]:
```suggestion
  meta = {
    homepage = "https://github.com/initialcommit-com/git-dummy";
    description = "Generate dummy Git repositories populated with the desired number of commits, branches, and structure";
    license = lib.licenses.gpl2;
    maintainers = with lib.maintainers; [ mathiassven ];
```

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+  dependencies = [
+    gitpython
+    typer
+    pydantic-settings
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/git-dummy/default.nix:34]: Does not work when cross-compiling #308283 (same for other package)

File: pkgs/development/python-modules/git-sim/default.nix [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+      --zsh <($out/bin/git-sim --show-completion zsh)
+
+    ln -s ${git-dummy}/bin/git-dummy $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "Visually simulate Git operations in your own repos with a single terminal command";
+    homepage = "https://initialcommit.com/tools/git-sim";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ mathiassven ];
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/git-sim/default.nix:57]:
```suggestion
  meta = {
    description = "Visually simulate Git operations in your own repos with a single terminal command";
    homepage = "https://initialcommit.com/tools/git-sim";
    license = lib.licenses.gpl2;
    maintainers = with lib.maintainers; [ mathiassven ];
```

File: pkgs/development/python-modules/git-sim/default.nix (hunk 7 above)
Comment (FliegendeWurst) [reply pkgs/development/python-modules/git-sim/default.nix:50]: In that case, why include git-dummy in the output for git-sim?

File: pkgs/development/python-modules/manim/default.nix [hunk 12]
```diff
@@ -205,6 +205,9 @@ buildPythonPackage rec {
   postPatch = ''
     substituteInPlace pyproject.toml \
       --replace "--no-cov-on-fail --cov=manim --cov-report xml --cov-report term" ""
+
+    substituteInPlace manim/_config/default.cfg \
+      --replace "ffmpeg_executable = ffmpeg" "ffmpeg_executable = ${ffmpeg}/bin/ffmpeg"
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/manim/default.nix:210]:
```suggestion
      --replace "ffmpeg_executable = ffmpeg" "ffmpeg_executable = ${lib.getExe ffmpeg}"
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/git-sim/default.nix [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  gitpython,
+  manim,
+  opencv4,
+  typer,
+  pydantic,
+  fonttools,
+  git-dummy,
+}:
+
+buildPythonPackage rec {
```
Comment (natsukium) [inline pkgs/development/python-modules/git-sim/default.nix:15]: I think this package is an application but not a library.
Could you move it to `by-name/gi/git-sim` from `python-modules/git-sim`?
Also, please use `buildPythonApplication` instead.

File: pkgs/development/python-modules/git-sim/default.nix [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+  src = fetchFromGitHub {
+    owner = "initialcommit-com";
+    repo = "git-sim";
+    rev = "v${version}";
+    hash = "sha256-4jHkAlF2SAzHjBi8pmAJ0TKkcLxw+6EdGsXnHZUMILw=";
+  };
+
+  pythonRemoveDeps = [ "opencv-python-headless" ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/git-sim/default.nix:28]: `build-system` is missing.

File: pkgs/development/python-modules/git-sim/default.nix [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+  propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/git-sim/default.nix:29]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    installShellCompletion --cmd git-dummy \
+      --bash <($out/bin/git-dummy --show-completion bash) \
+      --fish <($out/bin/git-dummy --show-completion fish) \
+      --zsh <($out/bin/git-dummy --show-completion zsh)
+  '';
+
```
Comment (natsukium) [inline pkgs/development/python-modules/git-dummy/default.nix:40]: Could you enable tests?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

File: pkgs/development/python-modules/git-dummy/default.nix [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/initialcommit-com/git-dummy";
+    description = "Generate dummy Git repositories populated with the desired number of commits, branches, and structure";
+    license = licenses.gpl2;
```
Comment (natsukium) [inline pkgs/development/python-modules/git-dummy/default.nix:44]:
```suggestion
    license = licenses.gpl2Only;
```
`lincenses.gpl2` is deprecated.

Could you ask the upstream to confirm the licence?
This line may be wrong.
https://github.com/initialcommit-com/git-dummy/blob/dc85f631b06385c0f03d112ba3c0cb35162c2ab1/setup.py#L18

File: pkgs/development/python-modules/git-sim/default.nix [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+      --fish <($out/bin/git-sim --show-completion fish) \
+      --zsh <($out/bin/git-sim --show-completion zsh)
+
+    ln -s ${git-dummy}/bin/git-dummy $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "Visually simulate Git operations in your own repos with a single terminal command";
+    homepage = "https://initialcommit.com/tools/git-sim";
+    license = licenses.gpl2;
```
Comment (natsukium) [inline pkgs/development/python-modules/git-sim/default.nix:56]:
```suggestion
    license = licenses.gpl2Only;
```

**Review by natsukium — DISMISSED** [review]

File: pkgs/development/python-modules/git-sim/default.nix (hunk 7 above)
Comment (MathiasSven) [reply pkgs/development/python-modules/git-sim/default.nix:50]: It was to honour upstream's expectation that installing `git-sim` includes `git-dummy`

File: pkgs/development/python-modules/git-dummy/default.nix (hunk 17 above)
Comment (MathiasSven) [reply pkgs/development/python-modules/git-dummy/default.nix]: I have opened an issue to confirm, but I believe it is meant to be GPLv2 https://github.com/initialcommit-com/git-dummy/issues/5

File: pkgs/development/python-modules/git-dummy/default.nix (hunk 16 above)
Comment (MathiasSven) [reply pkgs/development/python-modules/git-dummy/default.nix:43]: The tests are currently broken, but I have added the parts to be able to run the tests.

File: pkgs/development/python-modules/git-dummy/default.nix (hunk 17 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/git-dummy/default.nix]: Its gpl2 on master now https://github.com/initialcommit-com/git-sim/blob/main/pyproject.toml#L27

**Pull request merged**